File: Data Structures & Algorithms/sum-of-two-integers/submission-0.java
class Solution {
    public int getSum(int a, int b) {
        // xor
        // carry 1's need to shift left by 1, (a & b) << 1
        // then add them together (xor and carry), and repeat, no carry means we are done

        while (b != 0) {
            int temp = (a & b) << 1;
            a = a ^ b;
            b = temp;
        }
        return a;
    }
}

File: Data Structures & Algorithms/plus-one/submission-0.py
class Solution:
    def plusOne(self, digits: List[int]) -> List[int]:
        # T: O(n), S: O(1)
        
        digits.reverse()
        carry, i = 1, 0

        while carry:
            if i < len(digits):
                if digits[i] == 9:
                    digits[i] = 0
                else:
                    digits[i] += 1
                    carry = 0
            else:
                digits.append(1)
                carry = 0
            i += 1
        digits.reverse()
        return digits